internal/service/currency-service.go

```diff
@@ -9,49 +9,52 @@ import (
 	"net/url"
 	"time"
 
+	"genesis-currency-api/pkg/config"
+
 	"genesis-currency-api/pkg/dto"
 	"genesis-currency-api/pkg/errors"
 	"genesis-currency-api/pkg/util/date"
-	"github.com/spf13/viper"
 )
 
-type CurrencyService struct {
+type CurrencyServiceImpl struct {
 	currencyInfo dto.CurrencyInfoDTO
+	cnf          config.CurrencyServiceConfig
 }
 
-// NewCurrencyService is a factory function for CurrencyService
-func NewCurrencyService() *CurrencyService {
+// NewCurrencyServiceImpl is a factory function for CurrencyServiceImpl
+func NewCurrencyServiceImpl(cnf config.CurrencyServiceConfig) *CurrencyServiceImpl {
 	// A cache value for 3rd party API response.
 	var currencyInfo dto.CurrencyInfoDTO
-	c := &CurrencyService{
+	c := &CurrencyServiceImpl{
```

Comment:
As for me, this func violates S (single responsibility from SOLID) 

* create instance
* update rates

This is constructor, 


```go

type CurrencyServiceImpl struct {
	currencyInfo dto.CurrencyInfoDTO
	cnf          config.CurrencyServiceConfig
}

func NewCurrencyServiceImpl(currencyInfo dto.CurrencyInfoDTO, cnf config.CurrencyServiceConfig) *CurrencyServiceImpl {
	return &CurrencyServiceImpl{currencyInfo: currencyInfo, cnf: cnf}
}

```

Reply:
Fixed this issue in the future PR HW3.